src/main/java/com/denizenscript/ddiscordbot/commands/DiscordInteractionCommand.java

```diff
@@ -0,0 +1,564 @@
+        List<Command> retrievedCmds = null;
+        if (group == null) {
+            retrievedCmds = client.retrieveCommands().complete();
+        }
+        else {
+            retrievedCmds = group.getGuild().retrieveCommands().complete();
+        }
+        String matchString = CoreUtilities.toLowerCase(name.asString());
+        Command bestMatch = null;
+        for (Command cmd : retrievedCmds) {
+            String commandName = CoreUtilities.toLowerCase(cmd.getName());
+            if (matchString.equals(commandName)) {
+                bestMatch = cmd;
+                break;
+            }
+            if (commandName.contains(matchString)) {
+                bestMatch = cmd;
+            }
+        }
+        if (bestMatch == null) {
+            Debug.echoError(scriptEntry.getContext(), "Invalid command name!");
+            scriptEntry.setFinished(true);
+            return null;
+        }
+        return bestMatch;
+    }
+
+    void addPrivileges(ScriptEntry scriptEntry, boolean enable, ListTag actionFor, List<CommandPrivilege> privileges) {
+        for (String item : actionFor) {
+            CommandPrivilege result = null;
+            if (item.startsWith("discorduser@")) {
+                result = CommandPrivilege.enable(DiscordUserTag.valueOf(item, scriptEntry.getContext()).getUser());
+            } 
+            else if (item.startsWith("discordrole@")) {
+                result = CommandPrivilege.enable(DiscordRoleTag.valueOf(item, scriptEntry.getContext()).role);
+            }
+            if (result != null) privileges.add(result);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public void execute(ScriptEntry scriptEntry) {
+        ElementTag id = scriptEntry.getElement("id");
+        ElementTag instruction = scriptEntry.getElement("instruction");
+        ElementTag commandInstruction = scriptEntry.getElement("command_instruction");
+        DiscordGroupTag group = scriptEntry.getObjectTag("group");
+        ElementTag name = scriptEntry.getElement("name");
+        ElementTag description = scriptEntry.getElement("description");
+        List<MapTag> options = (List<MapTag>) scriptEntry.getObjectTag("options");
+        ElementTag enabled = scriptEntry.getElement("enabled");
+        ListTag enableFor = scriptEntry.getObjectTag("enable_for");
+        ListTag disableFor = scriptEntry.getObjectTag("disable_for");
+        DiscordInteractionTag interaction = scriptEntry.getObjectTag("interaction");
+        ElementTag ephermal = scriptEntry.getElement("ephermal");
+        ElementTag attachFileName = scriptEntry.getElement("attach_file_name");
+        ElementTag attachFileText = scriptEntry.getElement("attach_file_text");
+        List<ListTag> rows = (List<ListTag>) scriptEntry.getObjectTag("rows");
+        ElementTag message = scriptEntry.getElement("message");
+        if (scriptEntry.dbCallShouldDebug()) {
+            Debug.report(scriptEntry, getName(), id.debug()
```

Comment:
Refer to source of any modern commands, or any recent comments to commands - the new `Debug.report` input format is much cleaner, especially for cases like this mess